pkg/builder/webhook.go

```diff
@@ -251,3 +270,11 @@ func generateValidatePath(gvk schema.GroupVersionKind) string {
 	return "/validate-" + strings.ReplaceAll(gvk.Group, ".", "-") + "-" +
 		gvk.Version + "-" + strings.ToLower(gvk.Kind)
 }
+
+func generateCustomPath(customPath string) string {
```

Comment:
I don't think silently ignoring or changing the path that was passed in is a good idea, we should instead return an error.

Also, please compile the regex at the package level rather than inside this func.